.github/workflows/test.yaml

```diff
@@ -20,17 +20,22 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Build image
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+
+      - name: Install testing requirements
         run: |
-          docker build -t jupyter-remote-desktop-proxy .
+          pip install -r dev-requirements.txt
+          # Install websocat, needed for integration tests
+          wget -q https://github.com/vi/websocat/releases/download/v1.12.0/websocat.x86_64-unknown-linux-musl \
+            -O /usr/local/bin/websocat
+          chmod +x /usr/local/bin/websocat
 
-      - name: Smoke test image
+      - name: Run Integration tests
         run: |
-          docker run -d -p 8888:8888 -e JUPYTER_TOKEN=secret jupyter-remote-desktop-proxy
-          sleep 10
-          curl 'http://localhost:8888/desktop/?token=secret' | grep 'Jupyter Remote Desktop Proxy'
-          # Test if the built JS file is present in the image
-          curl 'http://localhost:8888/desktop/dist/viewer.js?token=secret' > /dev/null
```

Comment:
These curl tests have been converted into a pytest as well, and in fact actually caught the fact that the second viewer.js test doesn't actually work (wrong URL in the test, URL has been corrected because pytest caught it.)